frontend/public/components/modals/impersonate-user-modal.tsx

```diff
@@ -0,0 +1,366 @@
+              readOnly={isUsernameReadonly}
+              placeholder={t('public~Enter a username')}
+              data-test="username-input"
+              validated={usernameError ? 'error' : 'default'}
+            />
+            {usernameError && (
+              <FormHelperText>
+                <HelperText>
+                  <HelperTextItem variant="error" icon={<ExclamationCircleIcon />}>
+                    {usernameError}
+                  </HelperTextItem>
+                </HelperText>
+              </FormHelperText>
+            )}
+          </FormGroup>
+
+          <FormGroup
+            label={
+              <>
+                {t('public~Groups')}
+                <FieldLevelHelp>
+                  {t('public~The groups to impersonate the user with')}
+                </FieldLevelHelp>
+              </>
+            }
+            fieldId="impersonate-groups"
+          >
+            <Select
+              id="impersonate-groups"
+              isOpen={isGroupSelectOpen}
+              onOpenChange={(open) => setIsGroupSelectOpen(open)}
+              onSelect={handleGroupSelect}
+              toggle={toggle}
+              isScrollable
+              maxMenuHeight="300px"
+              popperProps={{
+                enableFlip: false,
+                direction: 'down',
+              }}
+            >
+              <SelectList id="impersonate-groups-listbox">
+                {filteredGroups.length === 0 ? (
+                  <SelectOption isDisabled>{t('public~No results found')}</SelectOption>
+                ) : (
+                  <>
+                    <SelectOption
+                      key="__select_all__"
+                      value="__select_all__"
+                      isSelected={areAllFilteredGroupsSelected}
+                    >
+                      {t('public~Select all')}
+                    </SelectOption>
+                    {filteredGroups.map((group) => (
+                      <SelectOption
+                        key={group}
+                        value={group}
+                        isSelected={selectedGroups.includes(group)}
+                      >
+                        {group}
+                      </SelectOption>
+                    ))}
+                  </>
+                )}
+              </SelectList>
+            </Select>
+
+            {selectedGroups.length > 0 && (
+              <Flex spaceItems={{ default: 'spaceItemsSm' }} className="pf-v5-u-mt-sm">
+                {visibleGroups.map((group) => (
+                  <FlexItem key={group}>
+                    <Label onClose={() => handleGroupRemove(group)} color="blue">
+                      {group}
+                    </Label>
+                  </FlexItem>
+                ))}
+                {!showAllGroups && remainingCount > 0 && (
+                  <FlexItem>
+                    <Label
+                      color="grey"
+                      className="pf-v5-u-cursor-pointer"
```

Comment:
do not use pf-v5, use `isClickable` prop

https://www.patternfly.org/components/label/#label-with-custom-render